P2PEnabledConference.js

```diff
@@ -0,0 +1,899 @@
+        /**
+         * A delay given in seconds, before the conference switches back to P2P
+         * after the 3rd participant has left.
+         * @type {number}
+         */
+        this.backToP2PDelay = isNaN(delay) ? 5 : delay;
+        logger.info("backToP2PDelay: " + this.backToP2PDelay);
+
+        /**
+         * If set to <tt>true</tt> it means the P2P ICE is no longer connected.
+         * When <tt>false</tt> it means that P2P ICE (media) connection is up
+         * and running.
+         * @type {boolean}
+         */
+        this.isP2PConnectionInterrupted = false;
+        /**
+         * Flag set to <tt>true</tt> when P2P session has been established
+         * (ICE has been connected).
+         * @type {boolean}
+         */
+        this.p2pEstablished = false;
+        /**
+         * Fake <tt>ChatRoom</tt> passed to {@link peerToPeerSession}.
+         * @type {FakeChatRoomLayer}
+         */
+        this.peerToPeerFakeRoom = null;
+        /**
+         * A JingleSession for the direct peer to peer connection.
+         * @type {JingleSessionPC}
+         */
+        this.peerToPeerSession = null;
+    }
+
+    /**
+     * Accept incoming P2P Jingle call.
+     * @param {JingleSessionPC} jingleSession the session instance
+     * @param {jQuery} jingleOffer a jQuery selector pointing to 'jingle' IQ
+     * element.
+     * @private
+     */
+    _acceptP2PIncomingCall (jingleSession, jingleOffer) {
+
+        jingleSession.setSSRCOwnerJid(this.room.myroomjid);
+
+        // Accept the offer
+        this.peerToPeerSession = jingleSession;
+        // FIXME .P2P should be set initially in strophe.jingle.js
+        this.peerToPeerSession.isP2P = true;
+        this.peerToPeerFakeRoom = this._createFakeRoom(false);
+        this.peerToPeerSession.initialize(
+            false /* initiator */, this.peerToPeerFakeRoom, this.rtc);
+
+        const localTracks = this.getLocalTracks();
+
+        logger.debug("Adding " + localTracks + " to P2P...");
+        this.peerToPeerSession.addLocalTracks(localTracks).then(
+            () => {
+                logger.debug("Add " + localTracks + " to P2P done!");
+                this.peerToPeerSession.acceptOffer(
+                    jingleOffer,
+                    () => {
+                        logger.debug("Got RESULT for P2P 'session-accept'");
+                    },
+                    (error) => {
+                        logger.error(
+                            "Failed to accept incoming P2P Jingle session",
+                            error);
+                    }
+                );
+            },
+            (error) => {
+                logger.error(
+                    "Failed to add " + localTracks + " to the P2P connection",
+                    error);
+            });
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    _addLocalTrackAsUnmute (track) {
+        const all = [super._addLocalTrackAsUnmute(track)];
```

Comment:
could we use a more descriptive variable name here?  at first i thought this was referring to 'all local tracks', but i see below it's actually a set of promises